### PR TITLE
update Defun doc to clarify that definitions are frozen at first .run call

### LINKED ISSUE #6804: Defun broken: errors_impl.NotFoundError: Op type not registered
This code fails:

```
import tensorflow as tf
from tensorflow.python.framework import function

@function.Defun(tf.float32)
def custom_op(x):
    return x

x = tf.Variable(1, dtype=tf.float32)

sess = tf.Session()
sess.run(tf.global_variables_initializer())
print(sess.run(custom_op(x)))

sess.close()
```

Exception:

    tensorflow.python.framework.errors_impl.NotFoundError: Op type not registered 'custom_op_da39a3ee'

### PR DESCRIPTION
**Pull request by yaroslavvb**
Fixes #6804

### CODE REVIEW

**Review by drpngx — APPROVED** [review]

**Comment by drpngx** [comment]
Jenkins, test this please.

File: tensorflow/python/framework/function.py [hunk 1]
```diff
@@ -748,6 +748,9 @@ def foo(x, y):
   default graph and adds the definition of the function into the
   default graph. Because the addition of the function into the graph
   is deferred, the decorator can be used anywhere in the program.
+  
+  Definitions of function is frozen for the session after first session.run
```
Comment (zffchen78) [inline tensorflow/python/framework/function.py]: How about

Definitions of functions are frozen in a graph as soon as the graph is used to create a session.
Therefore, nodes using the function must be created in the graph before the corresponding session
is created.  

Maybe then add an example.

**Review by zffchen — CHANGES_REQUESTED** [review]
Thank you for the update, Yaralsav.

File: tensorflow/python/framework/function.py (hunk 1 above)
Comment (yaroslavvb) [inline tensorflow/python/framework/function.py]: That doesn't seem to match current behavior -- I can create a session, then create function node, and then issue first session.run and it works, still want me to change it to that?

**Comment by drpngx** [comment]
We need to wait two weeks on that one, sorry about that.

**Comment by rmlarsen** [comment]
@zffchen78 @drpngx Do you have additional comments? Can we merge this or are we still in the two week waiting period mentioned above?

**Comment by yaroslavvb** [comment]
changed to the wording suggested by zf

**Comment by rmlarsen** [comment]
Thanks @yaroslavvb. @zffchen78 can you take another look, please?

**Review by zffchen — APPROVED** [review]

**Comment by drpngx** [comment]
Jenkins, test this please.

**Pull request merged**